From 3c825ed309b3b05aff3ecf1f1f11d476d2864cbc Mon Sep 17 00:00:00 2001
From: Tomi Ollila <tomi.ollila@iki.fi>
Date: Tue, 18 Jun 2024 17:15:27 +0300
Subject: [PATCH 4/7] set LANG=C in environment

in order to be able to open filenames w/ 8-bit (e.g. w/ utf-8) chars
---
 src/main.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/main.cpp b/src/main.cpp
index c29842d..80c72c8 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -51,6 +51,7 @@ XX_NAMESPACE_USING
 
 extern char** environ;
 static char s_env_string_LC_ALL[] = "LC_ALL=C";
+static char s_env_string_LANG[] = "LANG=C";
 
 //------------------------------------------------------------------------------
 //
@@ -60,6 +61,8 @@ int main( int argc, char** argv, char** envp )
 
    // Override user locale or xxdiff could not parse localized diff output...
    putenv(s_env_string_LC_ALL);
+   // Ditto, override LANG, for filenames with 8-bit chars to work...
+   putenv(s_env_string_LANG);
 
    int retval = 2; // errors.
    try {
-- 
2.49.0

